data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexConfiguration.java

```diff
@@ -32,6 +32,7 @@ public class IndexConfiguration {
     public static final String NUM_REPLICAS = "number_of_replicas";
     public static final String BULK_SIZE = "bulk_size";
     public static final String DOCUMENT_ID_FIELD = "document_id_field";
+    public static final String ROUTING_ID_FIELD = "routing_id_field";
```

Comment:
I don't think this is an "id" per se. The index document API calls this simply `routing`. Please update to `routing_field`.